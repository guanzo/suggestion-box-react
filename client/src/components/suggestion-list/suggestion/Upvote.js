import React, { Component } from 'react';
import { connect } from 'react-redux'
import { toggleUpvote } from '@/store/suggestions'
import classNames from 'classnames'

class Upvote extends Component {
    render() {
        console.log(this.props)
        let { hasUpvoted, votesLength } = this.props
        let thumbsClass = classNames('fa m-r-5',
        hasUpvoted ? 'fa-thumbs-up':'fa-thumbs-o-up',
        { 
            'has-text-success': votesLength > 0 
        })

        return (
        <div class="flex">
            <i onClick={this.props.toggleUpvote} class={thumbsClass}></i>
            <div class="is-size-7">{votesLength.toLocaleString()}</div>
        </div>
        );
    }
}

const mapDispatchToProps = (dispatch,ownProps) => {
    return {
        toggleUpvote: ()=> dispatch(toggleUpvote(ownProps.hasUpvoted))
    }
}
const Upvote_C = connect(null, mapDispatchToProps)(Upvote)

export default Upvote_C;

