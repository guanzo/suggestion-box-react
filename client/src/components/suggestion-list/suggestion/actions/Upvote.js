import React, { PureComponent } from 'react';
import { connect } from 'react-redux'
import { toggleUpvote } from '@/store/suggestions'
import classNames from 'classnames'

class Upvote extends PureComponent {
    render() {
		let { hasUpvoted, votesLength, broadcasterUpvoted, channel } = this.props
        let iconClass = classNames('fa m-r-5',
        hasUpvoted ? 'fa-thumbs-up':'fa-thumbs-o-up',
        { 
            'has-text-primary': votesLength > 0 
        })

        return (
        <div className="flex align-center justify-start">
            <i onClick={this.props.toggleUpvote} className={iconClass}></i>
            <div className="is-size-7">{votesLength.toLocaleString()}</div>
			<p className="m-l-10 is-size-7">{ broadcasterUpvoted ? `${channel.channelName} likes this` : null }</p>
        </div>
        );
    }
}

const mapDispatchToProps = (dispatch,ownProps) => {
    return {
        toggleUpvote: ()=> dispatch(toggleUpvote(ownProps))
    }
}
export default connect(null, mapDispatchToProps)(Upvote)


