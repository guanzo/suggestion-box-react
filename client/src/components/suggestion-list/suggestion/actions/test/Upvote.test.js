
import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {Upvote} from '../Upvote';
const { 
	SORT_VOTES, SORT_NEW, LIST_APPROVED, LIST_PENDING
} = require('@shared/suggestion-util')

function mockProps(){
	return {
		hasUpvoted: true,
		votesLength: 1,
		broadcasterUpvoted: true,
		channel:{
			channelName: 'guanzo'
		},
		currentUser: {
			isAdmin: false
		}
	}
}

describe('<Upvote />',()=>{
	it('renders correctly',()=>{
		const component = renderer.create(<Upvote {...mockProps()}/>)
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		/* tree.broadcasterUpvoted = false
		tree = component.toJSON();
		expect(tree).toMatchSnapshot(); */
	})
})