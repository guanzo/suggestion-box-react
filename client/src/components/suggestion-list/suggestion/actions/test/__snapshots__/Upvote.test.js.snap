// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Upvote /> renders correctly 1`] = `
<div
  className="upvote flex align-center m-r-5"
>
  <span
    className="icon"
  >
    <i
      className="fa fa-thumbs-up has-text-primary"
      onClick={undefined}
      style={
        Object {
          "cursor": "pointer",
        }
      }
      title=""
    />
  </span>
  <div
    className="is-size-7"
  >
    1
  </div>
</div>
`;
